util/src/main/java/com/google/cloud/hadoop/util/GcsClientEventBusFactory.java

```diff
@@ -0,0 +1,18 @@
+package com.google.cloud.hadoop.util;
+
+import com.google.common.eventbus.EventBus;
+
+/** Event Bus class */
+public class GcsClientEventBusFactory {
+
+  /** Hold the instance of the event bus here */
+  private static EventBus eventBus = new EventBus();
+  /**
+   * Returning the instance of the event bus here
+   *
+   * @return instance of eventBus
+   */
+  public static EventBus onResponseReceived() {
```

Comment:
Let us not expose the EventBus object. Why not have a one method for each type of event and call the post() from that.